opencog/query/PatternMatchEngine.cc

```diff
@@ -57,9 +57,10 @@ static inline bool log(const Handle& h)
 
 static inline void logmsg(const char * msg, const Handle& h)
 {
-	LAZY_LOG_FINE << msg << (h == nullptr ?
-	                         std::string("(invalid handle)") :
-	                         h->toShortString());
+	LAZY_LOG_FINE << msg << std::endl
```

Comment:
you should not put the linefeed here, it wrecks some of the message formats .. e.g. when printing two aligned sequences of atoms on the same line... for example, the ones immediately below.   It is more important to have log messages that are easy to read, than to make them pretty.